realm/src/androidTest/java/io/realm/RealmInMemoryTest.java

```diff
@@ -0,0 +1,215 @@
+            fail("Realm.getInstance should fail with illegal argument");
+        } catch (IllegalArgumentException iae) { //NOPMD
+        }
+
+        // Create on-disk-realm first, then create in-mem-realm with the same name
+        testRealm.close();
+        testRealm = null;
+        Realm realmOnDisk = Realm.getInstance(
+                new RealmConfiguration.Builder(getContext()).name(IDENTIFIER).build());
+        try {
+            testRealm = Realm.getInMemoryInstance(getContext(), IDENTIFIER);
+            fail("Realm.getInMemoryInstance should fail with illegal argument");
+        } catch (IllegalArgumentException iae) { //NOPMD
+        }
+
+        realmOnDisk.close();
+    }
+
+    public void testDelete() {
+        RealmConfiguration configuration = testRealm.getConfiguration();
+        try {
+            Realm.deleteRealm(configuration);
+            fail("Realm.deleteRealm should fail with illegal state");
+        } catch (IllegalStateException iae) { //NOPMD
+        }
+
+        // Nothing should happen when delete a closed in-mem-realm.
+        testRealm.close();
+        testRealm = null;
+        Realm.deleteRealm(configuration);
+    }
+
+    public void testWriteCopyTo() throws IOException {
+        String fileName = IDENTIFIER + ".realm";
+        String encFileName = IDENTIFIER + ".enc.realm";
+        byte[] key = TestHelper.getRandomKey();
+
+        Realm.deleteRealm(new RealmConfiguration.Builder(getContext()).name(fileName).build());
+        Realm.deleteRealm(new RealmConfiguration.Builder(getContext()).name(encFileName).build());
+
+        testRealm.beginTransaction();
+        Dog dog = testRealm.createObject(Dog.class);
+        dog.setName("DinoDog");
+        testRealm.commitTransaction();
+
+        // Test a normal realm file
+        testRealm.writeCopyTo(new File(getContext().getFilesDir(), fileName));
+        Realm onDiskRealm = Realm.getInstance(new RealmConfiguration.Builder(getContext()).name(fileName).build());
+        assertEquals(onDiskRealm.allObjects(Dog.class).size(), 1);
+        onDiskRealm.close();
+
+        // Test a encrypted realm file
+        testRealm.writeEncryptedCopyTo(new File(getContext().getFilesDir(), encFileName), key);
+        onDiskRealm = Realm.getInstance(new RealmConfiguration.Builder(getContext()).name(encFileName)
+                .encryptionKey(key).build());
+        assertEquals(onDiskRealm.allObjects(Dog.class).size(), 1);
+        onDiskRealm.close();
+        // Test with a wrong key
+        try {
+            Realm.getInstance(new RealmConfiguration.Builder(getContext()).name(encFileName)
+                    .encryptionKey(TestHelper.getRandomKey(42)).build());
+            fail("Realm.getInstance should fail with illegal argument");
+        } catch (IllegalArgumentException iae) { //NOPMD
+        }
+    }
+
+    public void testCreateWithConfiguration() {
```

Comment:
It is trying to test if a in-memory realm can be written to the disk with encryption and the encrytion does work by using a wrong key to open it.